dex/elrond_dex_farm/src/token_merge.rs

```diff
@@ -0,0 +1,224 @@
+elrond_wasm::imports!();
+elrond_wasm::derive_imports!();
+
+use farm_token::{FarmToken, FarmTokenAttributes};
+
+use super::config;
+
+use super::farm_token;
+
+#[derive(Clone, Copy)]
+pub struct ValueWeight<BigUint: BigUintApi> {
+    value: BigUint,
+    weight: BigUint,
+}
+
+#[elrond_wasm_derive::module]
+pub trait TokenMergeModule:
+    nft_deposit::NftDepositModule
+    + token_send::TokenSendModule
+    + farm_token::FarmTokenModule
+    + token_supply::TokenSupplyModule
+    + config::ConfigModule
+{
+    #[endpoint(mergeTokens)]
+    fn merge_tokens(
+        &self,
+        #[var_args] opt_accept_funds_func: OptionalArg<BoxedBytes>,
+    ) -> SCResult<()> {
+        let caller = self.blockchain().get_caller();
+        let attrs = self.get_merged_farm_token_attributes(&caller, Option::None)?;
+        let farm_token_id = self.farm_token_id().get();
+
+        self.burn_merge_tokens(&caller);
+        self.nft_create_tokens(&farm_token_id, &attrs.current_farm_amount, &attrs);
+        self.increase_nonce();
+
+        self.send_nft_tokens(
```

Comment:
I think instead of simple merge we could go directly with harvesting all positions, exiting farm with all positions, and entering it with the new accumulated tokens. This especially for FARM. separate merge can stay, but harvesting will be 99% of the times used.

Reply:
yes but we agreed on this way. it seems more intuitive and more practical from my side

Reply:
there would be a lot of redundant storage updates and a lot of things that keep of changing for no reason at all. we will easily run into gas issues when we will integrate this in the proxy contract, when merging proxy specific tokens

Reply:
it is ok. I was thinking only about the alternatives and about what the users want. We would save one transaction when the user wants to addToFarm and harvest as well. It can stay like this, however we might want further improvements which makes it easier for users to interact with the contracts, and makes them click less buttons, less transactions.
In most of the cases len of merged tokens will be 1. That is something we will try to ensure from Maiar Dex frontend

Reply:
if the frontend is made intuitive one transaction shouldn't cause problems. i'd trade that for a more atomic endpoint, that does only one thing